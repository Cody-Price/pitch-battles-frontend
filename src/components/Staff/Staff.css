.staff {
  width: 345px;
  height: 150px;
  background: url("./staff/staff.png");
  background-size: cover;
  margin: 3rem auto 5rem;
  image-rendering: pixelated;
  position: relative;
}

.clef {
  position: absolute;
  background: url("./staff/trebleclef.png");
  top: 25px;
  left: 52px;
  height: 117px;
  width: 48px;
  background-size: cover;
  image-rendering: pixelated;
}

.pitch {
  position: absolute;
  background: url("./staff/notehead3.png");
  width: 39px;
  height: 21px;
  image-rendering: pixelated;
  background-size: cover;
  left: 150px;
  top: 123px;
}

.pitch.below-space-one {
  top: 123px;
}
.pitch.line-one {
  top: 111px;
}

.pitch.space-one {
  top: 99px;
}

.pitch.line-two {
  top: 87px;
}

.pitch.space-two {
  top: 75px;
}

.pitch.line-three {
  top: 63px;
}

.pitch.space-three {
  top: 51px;
}

.pitch.line-four {
  top: 39px;
}

.pitch.space-four {
  top: 27px;
}

.pitch.line-five {
  top: 15px;
}

.pitch.above-space-one {
  top: 3px;
}

.pitch.above-space-two {
  background: url("./staff/notehead-one-ledger-above.png");
  top: -15px;
  width: 39px;
  height: 21px;
  image-rendering: pixelated;
  background-size: cover;
}

.pitch.above-line-one {
  background: url("./staff/notehead-one-ledger.png");
  top: -8px;
  width: 39px;
  height: 18px;
  image-rendering: pixelated;
  background-size: cover;
}

.pitch.below-line-one {
  background: url("./staff/notehead-one-ledger.png");
  top: 133px;
  width: 39px;
  height: 18px;
  image-rendering: pixelated;
  background-size: cover;
}

.pitch {
  background: url("./staff/notehead-two-ledger-below-space.png");
  top: 145px;
  width: 39px;
  height: 45px;
  image-rendering: pixelated;
  background-size: cover;
}
