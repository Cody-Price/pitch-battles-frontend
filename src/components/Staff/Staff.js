import React, { Component } from "react";
import NoteRow from "../NoteRow/NoteRow";

import "./Staff.css";

class Staff extends Component {
  render() {
    return (
      <div className="staff-buffer">
        <section className="staff">
          <div className="clef" />
          <div className={`pitch ${this.props.currentPitch}`} />
        </section>
        <NoteRow submitGuess={this.props.submitGuess} />
      </div>
    );
  }
}

export default Staff;
