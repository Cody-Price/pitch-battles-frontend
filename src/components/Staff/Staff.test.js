import React from "react";
import ReactDOM from "react-dom";
import Staff from "./Staff";
import Enzyme, { shallow } from "enzyme";
const EnzymeAdapter = require("enzyme-adapter-react-16");

Enzyme.configure({ adapter: new EnzymeAdapter() });

describe("Staff", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Staff />);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
