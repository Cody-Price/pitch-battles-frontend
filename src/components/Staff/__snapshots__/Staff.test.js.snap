// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Staff should match the snapshot 1`] = `
<div
  className="staff-buffer"
>
  <section
    className="staff"
  >
    <p
      className="correct-answer undefined"
    >
      correct!
    </p>
    <p
      className="incorrect-answer undefined"
    >
      incorrect...
    </p>
    <div
      className="clef undefined"
    />
    <div
      className="pitch undefined"
    />
  </section>
  <NoteRow />
</div>
`;
